serdes-native/src/main/java/io/ballerina/stdlib/serdes/Deserializer.java

```diff
@@ -0,0 +1,360 @@
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package io.ballerina.stdlib.serdes;
+
+import com.google.protobuf.ByteString;
+import com.google.protobuf.Descriptors.Descriptor;
+import com.google.protobuf.Descriptors.FieldDescriptor;
+import com.google.protobuf.DynamicMessage;
+import com.google.protobuf.InvalidProtocolBufferException;
+import io.ballerina.runtime.api.TypeTags;
+import io.ballerina.runtime.api.creators.TypeCreator;
+import io.ballerina.runtime.api.creators.ValueCreator;
+import io.ballerina.runtime.api.types.ArrayType;
+import io.ballerina.runtime.api.types.Field;
+import io.ballerina.runtime.api.types.RecordType;
+import io.ballerina.runtime.api.types.Type;
+import io.ballerina.runtime.api.types.UnionType;
+import io.ballerina.runtime.api.utils.StringUtils;
+import io.ballerina.runtime.api.values.BArray;
+import io.ballerina.runtime.api.values.BError;
+import io.ballerina.runtime.api.values.BMap;
+import io.ballerina.runtime.api.values.BObject;
+import io.ballerina.runtime.api.values.BString;
+import io.ballerina.runtime.api.values.BTypedesc;
+
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.Iterator;
+import java.util.Map;
+
+import static io.ballerina.stdlib.serdes.Constants.SERDES_ERROR;
+import static io.ballerina.stdlib.serdes.Utils.createSerdesError;
+
+/**
+ * Deserializer class to generate an object from a byte array.
+ */
+public class Deserializer {
+
+    static final String BYTE = "LiteralByteString";
+    static final String STRING = "String";
+    static final String FLOAT = "Float";
+    static final String DOUBLE = "Double";
+    static final String DYNAMIC_MESSAGE = "DynamicMessage";
+
+    static final String ATOMIC_FIELD_NAME = "atomicField";
+    static final String ARRAY_FIELD_NAME = "arrayfield";
+    static final String NULL_FIELD_NAME = "nullField";
+
+    static final String SCHEMA_NAME = "schema";
+
+    static final String UNION_FIELD_NAME = "unionelement";
+    static final String UNION_TYPE_IDENTIFIER = "ballerinauniontype";
+    static final String UNION_FIELD_SEPARATOR = "__";
+
+    static final String UNSUPPORTED_DATA_TYPE = "Unsupported data type: ";
+    static final String DESERIALIZATION_ERROR_MESSAGE = "Failed to Deserialize data: ";
+
+    /**
+     * Creates an anydata object from a byte array after deserializing.
+     *
+     * @param des  Deserializer object.
+     * @param encodedMessage Byte array corresponding to encoded data.
+     * @param dataType Data type of the encoded value.
+     * @return anydata object.
+     */
+    public static Object deserialize(BObject des, BArray encodedMessage, BTypedesc dataType) {
+        Descriptor schema = (Descriptor) des.getNativeData(SCHEMA_NAME);
+
+        Object object = null;
+        
+        try {
+            DynamicMessage dynamicMessage = generateDynamicMessageFromBytes(schema, encodedMessage);
+
```

Comment:
An extra empty line. Check for the other places as well. An extra line would be necessary when we want to separate two functions, of logical blocks (a loop from a if-else block), etc. Otherwise let's not add empty lines like this.